test/evp_test.c

```diff
@@ -1258,6 +1259,13 @@ static int pkey_test_ctrl(EVP_PKEY_CTX *pctx, const char *value)
     if (p != NULL)
         *p++ = 0;
     rv = EVP_PKEY_CTX_ctrl_str(pctx, tmpval, p);
+    if (p != NULL && rv <= 0 && rv != -2) {
```

Comment:
Hmmm...but this also assumes that the thing after the colon is even intended to be something that could have an oid...they don't all start with digest...although I can't think of a better way either.